pgrx/src/spi.rs

```diff
@@ -876,9 +875,9 @@ impl<'a> PreparedStatement<'a> {
     }
 }
 
-impl<'a: 'b, 'b> Query for &'b PreparedStatement<'a> {
+impl<'conn> Query<'conn> for &PreparedStatement<'conn> {
```

Comment:
Hmmm, previously we said `'a` outlives `'b`. This may fall prey to the wrong lifetime being inferred, but an easy fix if that turns out to be the case.

Reply:
The only way to create a `PreparedStatement` is through `SpiClient::prepare()` which can only happen within `Spi::connect()` and now as of 0e5b5ae48b54bcd9f856c836c6e90c00a11740c9 all the lifetime connections should be sorted such that it's also no possible to return a `PreparedStatement` from `Spi::connect(|c| ...)`.

That said, I added these bounds back (with a better name) because yes, unlike the `&str` case, the connection needs to "outlive" the prepared statement.  Per the Postgres docs (https://www.postgresql.org/docs/current/spi-spi-prepare.html), a prepared statement only lives as long as you're under `SPI_connect()`.